subiquity/models/tests/test_filesystem.py

```diff
@@ -78,3 +159,702 @@ def test_errors(self):
+        swappart.id = os.path.basename(swappart.path)
+        swapfs = self.fsm.add_filesystem(swappart, 'swap')
+        swapfs.id = "%s-%s" % (swappart.id, swapfs.fstype)
+        self.fsm.add_mount(swapfs, "")
+
+        config = self.fsm.render()
+        for item in config:
+            print(item)
+        mounts = [entry for entry in config if entry['type'] == 'mount']
+        self.assertEqual(1, len(mounts))
+        swapmount = mounts[0]
+        self.assertEqual(swapfs.id, swapmount['device'])
+
+    def test_render_ensures_swap_partition_w_flag_has_mount_nomount(self):
+        """render includes swap mount when partition.flag='swap' no mount"""
+        self.fsm.probe()
+        sda = self.fsm.all_disks()[0]
+        sda.id = os.path.basename(sda.path)
+        swappart = self.fsm.add_partition(disk=sda, size=(2 * MB),
+                                          flag='swap')
+        swappart.id = os.path.basename(swappart.path)
+        swapfs = self.fsm.add_filesystem(swappart, 'swap')
+        swapfs.id = "%s-%s" % (swappart.id, swapfs.fstype)
+
+        config = self.fsm.render()
+        for item in config:
+            print(item)
+        mounts = [entry for entry in config if entry['type'] == 'mount']
+        self.assertEqual(1, len(mounts))
+        swapmount = mounts[0]
+        self.assertEqual(swapfs.id, swapmount['device'])
+
+    def test_render_ensures_swap_partition_w_flag_has_mount_existing(self):
+        """render includes swap mount when partition.flag='swap' and mount"""
+        self.fsm.probe()
+        sda = self.fsm.all_disks()[0]
+        sda.id = os.path.basename(sda.path)
+        swappart = self.fsm.add_partition(disk=sda, size=(2 * MB),
+                                          flag='swap')
+        swappart.id = os.path.basename(swappart.path)
+        swapfs = self.fsm.add_filesystem(swappart, 'swap')
+        swapfs.id = "%s-%s" % (swappart.id, swapfs.fstype)
+        self.fsm.add_mount(swapfs, "")
+
+        config = self.fsm.render()
+        for item in config:
+            print(item)
+        mounts = [entry for entry in config if entry['type'] == 'mount']
+        self.assertEqual(1, len(mounts))
+        swapmount = mounts[0]
+        self.assertEqual(swapfs.id, swapmount['device'])
+
+    def test_render_ensures_swap_disk_has_mount_existing(self):
+        """"render includes swap mount when disk.fs='swap and mount"""
+        self.fsm.probe()
+        sda = self.fsm.all_disks()[0]
+        sda.id = os.path.basename(sda.path)
+        swapfs = self.fsm.add_filesystem(sda, 'swap')
+        swapfs.id = "%s-%s" % (sda.id, swapfs.fstype)
+        config = self.fsm.render()
+        for item in config:
+            print(item)
+        mounts = [entry for entry in config if entry['type'] == 'mount']
+        self.assertEqual(1, len(mounts))
+        swapmount = mounts[0]
+        self.assertEqual(swapfs.id, swapmount['device'])
+
+    def test_render_ensures_swap_disk_has_mount_nomount(self):
```

Comment:
wait how do all these test cases differ? Can you create some helpers to make them easier to read? Do we create Mounts for swap partitions or not? I think we only need to pick one, right?

Reply:
Right but I think we should (perhaps not in this branch) pick two of those and make sure those work. I was confused implementing this about the need for Mounts for swap partitions. Maybe we should implicitly create one (possibly not in model code) when we create a Swap fs? And we should just get the flag right.